serve.js

```diff
@@ -158,6 +168,15 @@ app.post("/add-courses", (req, res) => {
     });
 });
 
+// Perform the query entered into the admin portal
+app.post("/query", (req, res) => {
+    const query = req.body.query;
+    connection.query(query, (err, rows) => {
+        if (err) throw err;
+        res.end(JSON.stringify(rows));
+    });
+});
+
```

Comment:
The lack of validation on the backend makes this a vector for SQL injection attacks; an attacker can POST to `/query` with any query whatsoever. The static list of queries should be stored on the backend, and the query checked against that list; simply checking for the first word being "SELECT" also would not suffice because something like `SELECT * FROM courses; DROP TABLE courses` would still be allowed.